### PR TITLE
Default wait and demo

### LINKED ISSUE #129: Consider updating clx-demos
Cmucl has a different clx demo file in [demos.lisp](https://gitlab.common-lisp.net/cmucl/cmucl/blob/master/src/contrib/demos/demos.lisp).  This demos works significantly better than the clx-demos.lisp included here.  Many of the demos actually work such as qix and both towers of hanoi and greynetic.  These don't work for me with clx-demos.

The demos also allow the window manager to decorate the window so you can move the demo window around as well.  There's no window manager around the windows from clx-demos.lisp. 

I don't know if this works as well in other lisps, or if clx-demos works in other lisps. (Tried with ccl64 but nothing seems to work.  Don't know why.)

But if it works with other lisps, perhaps it can replace clx-demos.lisp or be added as a different demo.

### PR DESCRIPTION
**Pull request by dkochmanski**
- provide smaller poll time for dumb timeouts
- add ecl implementation using serve-event
- backport cmu demo (fixes #129)

### CODE REVIEW

**Comment by dkochmanski** [comment]
@rtoy I've backported some changes from CMU.

File: demo/clx-demos.lisp [hunk 1]
```diff
@@ -33,105 +39,89 @@
...
-     (sleep 3))))
-
-;;; DEMO is a hack to get by.  It should be based on creating a menu.  At
-;;; that time, *name-to-function* should be deleted, since this mapping will
-;;; be manifested in the menu slot name cross its action.  Also the
-;;; "Shove-bounce" demo should be renamed to "Shove bounce"; likewise for
-;;; "Fast-towers-of-Hanoi" and "Slow-towers-of-hanoi".
-;;;
+  (dolist (demo *demos*)
+    (funcall demo)
```
Comment (rtoy) [inline demo/clx-demos.lisp]: It would be useful to print out the demo name so you know what's being run, if you're not familiar with all the demos

Comment (dkochmanski) [reply demo/clx-demos.lisp]: window has a label (we have decoratiosn now) with demo name)

**Comment by dkochmanski** [comment]
As of what works and what doesn't:
works: qix, petal, *hanoi, recurrence, plaid
empty window: shove-bounce, bounce, bouncing-ball (I don't know what these are supposed to do)

This behavior is consistent on implementations I've tried (ecl, ccl, sbcl).

**Comment by dkochmanski** [comment]
also do-all-demos doesn't work as I would expect it to work. I'd remove the function whatsoever since we have menu now. what do you think?

**Comment by rtoy** [comment]
I think it's ok to commit what you have now.

For cmucl, qix just shows a bunch of lines, as if it ran super fast but didn't remove old lines.

Hanoi still has artifacts.

Shove-bounce doesn't move any windows around, and leaves a bunch of windows in the same place.  Bounce does nothing.

Bouncing-ball nows shows balls (bombs) dropping, but there are artifacts.

I looked and there are some differences in qix.  Don't know how significant these differences are.

And in case you're looking for more demos, the greynetic demo from demos.lisp works with cmucl. :-)

**Comment by dkochmanski** [comment]
Thanks. `greynetic` indeed works if taken from there. I'm merging this now then. as of bouncy and bouncy shove they work /most of the time/ for me, sometimes window refuses to move. probably a bug somewhere (i.e we query bottom position in racy condition with window creation or something in this spirit).

Thank you for working with me on this piece.

**Comment by rtoy** [comment]
No problem.  greynetic (from clx-demos) works in cmucl too!  It's a bit slow (I think *delay* should be lower for cmucl.  It's set to 0.01 in demos.lisp)  Greynetic definitely is faster.

One nice enhancement from cmu demos.lisp is that many demos use colors on a black background instead black on white.  (Greynetic, qix, petal use colors.)

**Pull request merged**